src/spikeinterface/generation/template_database.py

```diff
@@ -1,12 +1,133 @@
-from spikeinterface.core.template import Templates
 import zarr
+import numpy as np
+
+from spikeinterface.core.template import Templates
 
 
-def fetch_templates_from_database(dataset="test_templates.zarr") -> Templates:
+def fetch_template_dataset(dataset="test_templates.zarr") -> Templates:
+    """
+    Fetch a template dataset from the spikeinterface template database.
+    A dataset is a collection of templates with associated metadata for one specific recording.
 
+    Parameters
+    ----------
+    dataset : str, default: "test_templates"
+        The name of the dataset to fetch.
+        The dataset must be available in the spikeinterface template database.
+
+    Returns
+    -------
+    Templates
+        _description_
+    """
     s3_path = f"s3://spikeinterface-template-database/{dataset}/"
     zarr_group = zarr.open_consolidated(s3_path, storage_options={"anon": True})
 
     templates_object = Templates.from_zarr_group(zarr_group)
 
     return templates_object
+
+
+def fetch_templates_info() -> "pandas.DataFrame":
```

Comment:
```suggestion
import functools
@functools.cache
def fetch_templates_info() -> "pandas.DataFrame":
```

https://docs.python.org/3/library/functools.html#functools.cache

Reply:
Ah, this will fail if people still use python 3.8 : /

We can use `functools.lru_cache(maxsize=1)` instead if you want to support that but I would rather move forward to the bright future of python 3. 9 as minimal.

Reply:
yes let's do that for the sorting analyzer release